### PR TITLE
Feature/tweet component fetch

### PR DESCRIPTION
**Pull request by Mykyta-Pysarev**
image

Now tweet opens with comments if you click on the date.

### CODE REVIEW

**Review by VladislavPokhyl — APPROVED** [review]

File: src/containers/ProfileTweets.tsx [hunk 1]
```diff
@@ -35,15 +35,15 @@ const ProfileTweets: React.FunctionComponent = () => {
 
   return (
     <>
-      {mockUsers && (
+      {/* {mockUsers && (
```
Comment (KirillSerg) [inline src/containers/ProfileTweets.tsx:38]: You don't rener it? And don't use the approach from similar components.

File: src/containers/TweetPage.tsx [hunk 2]
```diff
@@ -23,7 +23,7 @@ export const TweetPage: React.FC = (): React.ReactElement => {
       </Wrap>
       <div>
         Тут буде твіт (компонент)
-        <SingleTweet {...singleTweetData} />
+        {/* <SingleTweet {...singleTweetData} /> */}
```
Comment (KirillSerg) [inline src/containers/TweetPage.tsx:26]: What differents?

File: src/components/FeedSingleTweet.tsx [hunk 3]
```diff
@@ -118,7 +119,12 @@ const FeedSingleTweet = ({
               <div className="mx-1 text-secondary d-flex justify-content-center align-items-center">
                 ·
               </div>
-              <UnderlineHover className="text-secondary">
+              <UnderlineHover
```
Comment (KirillSerg) [inline src/components/FeedSingleTweet.tsx:122]: Did I understand correctly that when clicking on the date of creation, it redirects to a page with a specific tweet (by id)? Then why only when clicking on the date and not the entire tweet (the entire tweet field)

File: src/containers/TweetsWithReplies.tsx [hunk 4]
```diff
@@ -1,19 +1,23 @@
...
-  page: 1,
-  limit: 10,
-  hasNextPage: false,
-};
-
 export const TweetsWithReplies: React.FunctionComponent = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const posts = useSelector<RootState, RootState["tweets"]["feedTweets"]>(
+    (store) => store.tweets.feedTweets
+  );
```
Comment (KirillSerg) [inline src/containers/TweetsWithReplies.tsx:13]: can we use specific tweets already as below?
const posts = useSelector<RootState, RootState["tweets"]["feedReplies"]>(
    (store) => store.tweets.feedReplies
  );

File: src/components/SingleTweet.tsx [hunk 5]
```diff
@@ -11,8 +11,8 @@ import {
 import { useTranslation } from "react-i18next";
 import styled from "styled-components";
 
-import { Tweet } from "../types";
-import AttachedContent from "./AttachedContent";
+import { Tweet2 } from "../types";
+// import AttachedContent from "./AttachedContent";
```
Comment (VitaliiDenysiuk) [inline src/components/SingleTweet.tsx:15]:
```suggestion
// import AttachedContent from "./AttachedContent";
```
clean code

File: src/components/SingleTweet.tsx [hunk 6]
```diff
@@ -96,17 +96,15 @@ const IconBg = styled.div<IIconBg>`
...
-  tweetQuoteCount,
-  likeCount,
-}) => {
+const SingleTweet = ({
+  // _id,
+  author,
+  content,
+  createdAt,
+  // repliedTo,
+  // updatedAt,
```
Comment (VitaliiDenysiuk) [inline src/components/SingleTweet.tsx:105]:
```suggestion
  author,
  content,
  createdAt,
  updatedAt,
```
clean code
use updatedAt on form

File: src/components/SingleTweet.tsx [hunk 7]
```diff
@@ -194,17 +192,17 @@ const SingleTweet: React.FC<Tweet> = ({
             </div>
           </NameSection>
         </Author>
-        <Text className="py-2">{tweetText}</Text>
-        <AttachedContent />
+        <Text className="py-2">{content}</Text>
+        {/* <AttachedContent /> */}
         <Date className="border-bottom py-3 fw-bold">
-          {tweetDate} - {lelekaLink}
+          {createdAt} - "lelekaLink"
```
Comment (VitaliiDenysiuk) [inline src/components/SingleTweet.tsx:198]: use normal data format (dd.mm.yyyy hh:mm) 
maybe it's needded add updatedAt to

File: src/containers/ProfileLikes.tsx [hunk 8]
```diff
@@ -1,20 +1,23 @@
...
-  hasNextPage: false,
-};
-
 const Likes = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const posts = useSelector<RootState, RootState["tweets"]["feedTweets"]>(
+    (store) => store.tweets.feedTweets
+  );
+  const handleShowMore = () => {
+    return !posts.isLoading && dispatch(tweetsActions.fetchFeedTweets(posts));
```
Comment (VitaliiDenysiuk) [inline src/containers/ProfileLikes.tsx]:
```suggestion
    return !posts.isLoading && posts.hasNextPage && dispatch(tweetsActions.fetchFeedTweets(posts));
```

File: src/containers/ProfileMedia.tsx [hunk 9]
```diff
@@ -1,20 +1,23 @@
...
-  hasNextPage: false,
-};
-
 const Media = () => {
+  const dispatch = useDispatch<AppDispatch>();
+  const posts = useSelector<RootState, RootState["tweets"]["feedTweets"]>(
+    (store) => store.tweets.feedTweets
+  );
+  const handleShowMore = () => {
+    return !posts.isLoading && dispatch(tweetsActions.fetchFeedTweets(posts));
```
Comment (VitaliiDenysiuk) [inline src/containers/ProfileMedia.tsx]:
```suggestion
    return !posts.isLoading && posts.hasNextPage && dispatch(tweetsActions.fetchFeedTweets(posts));
```

File: src/containers/ProfileTweets.tsx [hunk 10]
```diff
@@ -1,10 +1,10 @@
 import React, { useEffect, useState } from "react";
 
-import SingleTweet from "../components/SingleTweet";
+// import SingleTweet from "../components/SingleTweet";
```
Comment (VitaliiDenysiuk) [inline src/containers/ProfileTweets.tsx:3]:
```suggestion

```
clean

File: src/containers/ProfileTweets.tsx [hunk 11]
```diff
@@ -1,10 +1,10 @@
...
 import MOCKTweets from "../MOCKS/tweets";
 import { Tweet } from "../types";
 import { Pagination } from "../types/mock-api-types";
-import InfiniteList from "./InfiniteList";
+// import InfiniteList from "./InfiniteList";
```
Comment (VitaliiDenysiuk) [inline src/containers/ProfileTweets.tsx:7]:
```suggestion

```
clean

File: src/containers/ProfileTweets.tsx [hunk 12]
```diff
@@ -35,15 +35,15 @@ const ProfileTweets: React.FunctionComponent = () => {
...
+      {/* {mockUsers && (
         <InfiniteList<Tweet>
           showMore={fetchData}
           data={mockUsers}
           itemComponent={(itemData) => (
             <SingleTweet key={itemData.id} {...itemData} />
           )}
         />
-      )}
+      )} */}
```
Comment (VitaliiDenysiuk) [inline src/containers/ProfileTweets.tsx:46]: clean

File: src/containers/TweetPage.tsx [hunk 13]
```diff
@@ -2,8 +2,8 @@ import React from "react";
 import { ArrowLeft } from "react-bootstrap-icons";
 import styled from "styled-components";
 
-import SingleTweet from "../components/SingleTweet";
-import { singleTweetData } from "../MOCKS/singleTweetPage";
+// import SingleTweet from "../components/SingleTweet";
+// import { singleTweetData } from "../MOCKS/singleTweetPage";
```
Comment (VitaliiDenysiuk) [inline src/containers/TweetPage.tsx:6]:
```suggestion

```
clean

File: src/containers/TweetsWithReplies.tsx [hunk 14]
```diff
@@ -1,19 +1,23 @@
...
+  const handleShowMore = () => {
+    return !posts.isLoading && dispatch(tweetsActions.fetchFeedTweets(posts));
```
Comment (VitaliiDenysiuk) [inline src/containers/TweetsWithReplies.tsx]:
```suggestion
    return !posts.isLoading && posts.hasNextPage && dispatch(tweetsActions.fetchFeedTweets(posts));
```

**Review by VitaliiDenysiuk — CHANGES_REQUESTED** [review]

File: src/redux/reducers/tweets.ts [hunk 15]
```diff
@@ -8,9 +8,11 @@ import instance from "../../services/api";
 import { LE, Tweet2 } from "../../types";
 import { Pagination } from "../../types/mock-api-types";
 
-interface TweetsStore {
+export interface TweetsStore {
+  currentTweet: LE<{ data?: Tweet2 }>;
   feedTweets: LE<Pagination<Tweet2>>;
   singleTweet: LE<object>;
+  feedReplies: LE<Pagination<Tweet2>>;
```
Comment (alexandrtovmach) [inline src/redux/reducers/tweets.ts]:
```suggestion
  currentTweetRelies: LE<Pagination<Tweet2>>;
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]
small changes only

File: src/redux/reducers/tweets.ts [hunk 16]
```diff
@@ -8,9 +8,11 @@ import instance from "../../services/api";
 import { LE, Tweet2 } from "../../types";
 import { Pagination } from "../../types/mock-api-types";
 
-interface TweetsStore {
+export interface TweetsStore {
+  currentTweet: LE<{ data?: Tweet2 }>;
   feedTweets: LE<Pagination<Tweet2>>;
   singleTweet: LE<object>;
+  currentTweetRelies: LE<Pagination<Tweet2>>;
```
Comment (alexandrtovmach) [inline src/redux/reducers/tweets.ts]:
```suggestion
  currentTweetReplies: LE<Pagination<Tweet2>>;
```

**Review by alexandrtovmach — CHANGES_REQUESTED** [review]

**Review by alexandrtovmach — APPROVED** [review]

**Pull request merged**